src/osut/osut.py

```diff
@@ -257,6 +256,50 @@ def each_cons(it, n):
         yield tuple(deq)
 
 
+def clamp(value, minimum, maximum) -> float:
+    """In-house alternative to Numpy's 'clip' (re: Ruby's 'clamp').
+
+    Args:
+        value (float):
+            A float-convertible value (to clip/clamp).
+        minimum (float):
+            Lower bound.
+        maximum (float):
+            Upper bound.
+
+    Returns:
+        float: Clamped value. Either value, min, max or '0' if invalid inputs.
+
+    """
+    try:
+        value = float(value)
+    except:
+        try:
+            minimum = float(minimum)
+            return minimum
+        except:
+            try:
+                maximum = float(maximum)
+                return maximum
+            except:
+                return 0.0
```

Comment:
No logged warnings or errors if invalid input. Will ultimately return `0.0` if all three parameters can't be converted to floating point numbers.